R/geom_recessions.R

```diff
@@ -196,41 +207,57 @@ geom_recessions <- function(xformat = "numeric",
 # internal function used to define recessions table for use
 build_recessions <- function(update_recessions){
   if(is.logical(update_recessions)){
+    # if TRUE
     if(update_recessions){
       message("Trying to update recessions...")
-      return(
-        tryCatch(
-          suppressWarnings(update_recessions(quietly = TRUE)),
-          error = function(cond){
-            message("Could not update recessions. Using built-in recessions table...")
-            return(recessions)
-          })
-        )
+      updated_recessions <- suppressWarnings(update_recessions(quietly = TRUE))
+
+      # If updated_recessions is returned as NULL, use the default table
+      if (is.null(updated_recessions)) {
+        message("Could not update recessions. Using built-in recessions table...")
+        return(recessions)
+      }
+
+      message("Successfully fetched from NBER")
+      return(updated_recessions)
+      # if FALSE
     } else {
       return(recessions)
     }
+  # if DATAFRAME
   }else if(is.data.frame(update_recessions)){
     # confirm that table has correct structure
     if(!identical(update_recessions[NA,][1,], recessions[NA,][1,])){
       message("Recession table may not have correct format (See `?update_recessions`). Attempting anyway...")
     }
     return(update_recessions)
+  # OTHERWISE
   }else{
-    message("`update_recessions` must be TRUE, FALSE, or a data table. Using built-in recessions table...")
+    message("`update_recessions` must be TRUE, FALSE, or a data frame. Using built-in recessions table...")
     return(recessions)
   }
 }
 
 
 # Internal function designed to filter the built-in recessions table
-filter_recessions <- function(min, max, xformat, recess_table){
+filter_recessions <- function(min, max, xformat, show_ongoing, recess_table){
   # Bind local variables to function
-  end_num <- start_num <- end_date <- start_date <- end <- start <- NULL
+  end_num <- start_num <- end_date <- start_date <- end <- start <- ongoing <- NULL
 
   # unwrap recess_table from list
   recess_table <- recess_table[[1]]
 
-  # filter recessions correctly, based on xformat
+  # Add numeric version of dates
+  recess_table <- recess_table %>%
+    mutate(
+      start_num = lubridate::decimal_date(start_date),
+      end_num = lubridate::decimal_date(end_date))

```

Comment:
Thinking more about this, if this does not happen in update_recessions like it used to, it should only happen if it needs to--eg if xscale is numeric. It would be easy to move this down a few lines into the if statement, and mutate rather than rename to get to start and end. 

Also if this is here we probably need to import and require lubridate? Not sure why check doesn't catch this.